packages/node-client/lib/index.ts

```diff
@@ -733,6 +734,66 @@ export class Nango {
         return response.data;
     }
 
+    /**
+     * Wait for connection to be created
+     * @param integrationKey - The integration key
+     * @param userId - The user ID
+     * @param options - Optional configuration (signal for cancellation, maxAttempts, initialDelayMs, delayMs)
+     * @returns A promise that resolves when the connection is created
+     * @throws {Error} If the wait is aborted via signal or times out
+     */
+    public async waitForConnection(
+        integrationKey: string,
+        userId: string,
+        options?: { signal?: AbortSignal; maxAttempts?: number; initialDelayMs?: number; delayMs?: number }
+    ): Promise<ApiPublicConnection> {
+        const maxAttempts = options?.maxAttempts ?? 30;
+        const initialDelayMs = options?.initialDelayMs ?? 10000; // 10 seconds for initial auth flow
+        const delayBetweenAttemptsMs = options?.delayMs ?? 2000;
+        const signal = options?.signal;
+
+        if (signal?.aborted) {
+            throw new Error('Wait for connection was aborted');
+        }
+
+        for (let attempt = 0; attempt < maxAttempts; attempt++) {
+            if (signal?.aborted) {
+                throw new Error('Wait for connection was aborted');
+            }
+
+            const response = await this.listConnections(undefined, undefined, { endUserId: userId });
+            const connection = response.connections.find((conn) => conn.provider_config_key === integrationKey);
+
+            if (connection) {
+                return connection;
+            }
+
+            // Use longer delay for first attempt to account for auth flow
+            const currentDelay = attempt === 0 ? initialDelayMs : delayBetweenAttemptsMs;
+
+            await new Promise<void>((resolve, reject) => {
+                const timeout = setTimeout(resolve, currentDelay);
+
+                const onAbort = () => {
+                    clearTimeout(timeout);
+                    reject(new Error('Wait for connection was aborted'));
+                };
+
+                if (signal) {
+                    signal.addEventListener('abort', onAbort, { once: true });
+                }
```

Comment:
you can use the optional chaining here and remove the if: `signal?.addEventListener(...)`